### PR TITLE
Fix #4362

### LINKED ISSUE #4362: Add `aex` command to emulate hexpairs
This command must work like this:
```
$ r2 -a x86 -
> aex 40
```
This is `inc eax` for x86

### PR DESCRIPTION
**Pull request by Darredevil**

### CODE REVIEW

**Comment by Maijin** [comment]
There are some regressions, see here https://s3.amazonaws.com/archive.travis-ci.org/jobs/117851417/log.txt

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -3857,7 +3858,65 @@ static int cmd_anal(void *data, const char *input) {
...
+			}
+			r_anal_op_fini (&aop);
+			}
+			break;
+		case '?':
+		default:{
+			const char* help_msg[] = {
+			"Usage:", "ae", "[x] [...]",
+			"ae", " [expr]", "analyze opcode eval expression (see ao)",
+			"aex", " [hex]", "analyze opcode eval expression (see ao)",
```
Comment (radare) [inline libr/core/cmd_anal.c]: help message is wrong

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -200,7 +200,7 @@ static int process_input(RCore *core, const char *input, ut64* blocksize, char *
 }
 
 /* This function is not necessary anymore, but it's kept for discussion */
-static int process_input_pade(RCore *core, const char *input, char** hex, char **asm_arch, ut32 *bits) {
+R_API int r_core_process_input_pade(RCore *core, const char *input, char** hex, char **asm_arch, ut32 *bits) {
```
Comment (radare) [inline libr/core/cmd_print.c]: why?

File: libr/core/cmd_print.c [hunk 3]
```diff
@@ -77,7 +77,7 @@ static void cmd_print_eq_dict(RCore *core, int bsz) {
 	r_cons_printf ("block: %d  0x%x\n", bsz, bsz);
 }
 
-static void set_asm_configs(RCore *core, char *arch, ut32 bits, int segoff){
+R_API void r_core_set_asm_configs(RCore *core, char *arch, ut32 bits, int segoff){
```
Comment (radare) [inline libr/core/cmd_print.c]: why?

**Comment by radare** [comment]
ping

File: libr/core/cmd_print.c (hunk 3 above)
Comment (XVilka) [inline libr/core/cmd_print.c]: To be called from cmd_anal.c I think.

File: libr/core/cmd_print.c (hunk 2 above)
Comment (XVilka) [inline libr/core/cmd_print.c]: To be called from cmd_anal.c I think.

File: libr/core/cmd_anal.c [hunk 4]
```diff
@@ -3857,7 +3858,65 @@ static int cmd_anal(void *data, const char *input) {
...
+				str2 = calloc(sizeof(char), strlen(str)+1);
+				strcat(str2," ");
+				strcat(str2, str);
+				cmd_anal_esil (core, str2);
+			}
+			r_anal_op_fini (&aop);
+			}
+			break;
+		case '?':
+		default:{
```
Comment (XVilka) [inline libr/core/cmd_anal.c]: this on cause the error

File: libr/core/cmd_print.c (hunk 3 above)
Comment (Darredevil) [inline libr/core/cmd_print.c]: Exactly

**Comment by Darredevil** [comment]
@radare ready for merge

**Comment by radare** [comment]
Looks good! merged (see comments)

File: libr/core/cmd_anal.c [hunk 5]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+	case 'x':{ // "aex"
+		ut32 new_bits = -1;
+		int segoff, old_bits, pos = 0;
+		ut8 settings_changed = false;
+		char *new_arch = NULL, *old_arch = NULL, *hex = NULL;
+		old_arch = strdup (r_config_get (core->config, "asm.arch"));
+		old_bits = r_config_get_i (core->config, "asm.bits");
+		segoff = r_config_get_i (core->config, "asm.segoff");
+
+		if (input[0])
```
Comment (radare) [inline libr/core/cmd_anal.c]: use more brackets

File: libr/core/cmd_anal.c [hunk 6]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+		}
+
+		if (new_arch == NULL) new_arch = strdup (old_arch);
+		if (new_bits == -1) new_bits = old_bits;
+
+		if (strcmp (new_arch, old_arch) != 0 || new_bits != old_bits){
+			r_core_set_asm_configs (core, new_arch, new_bits, segoff);
+			settings_changed = true;
+		}
+		int ret, bufsz;
```
Comment (radare) [inline libr/core/cmd_anal.c]: do not define variables in the middle of the code

File: libr/core/cmd_anal.c [hunk 7]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+		RAnalOp aop = {0};
+		const char *str;
+		char *str2 = NULL;
+		bufsz = r_hex_str2bin (hex, (ut8*)hex);
+		ret = r_anal_op (core->anal, &aop, core->offset,
+			(const ut8*)hex, bufsz);
+		if (ret>0) {
+			str = R_STRBUF_SAFEGET (&aop.esil);
+			str2 = calloc(sizeof(char), strlen(str)+1);
```
Comment (radare) [inline libr/core/cmd_anal.c]: check if str2 is null

File: libr/core/cmd_anal.c [hunk 8]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+		}
+		int ret, bufsz;
+		RAnalOp aop = {0};
+		const char *str;
+		char *str2 = NULL;
+		bufsz = r_hex_str2bin (hex, (ut8*)hex);
+		ret = r_anal_op (core->anal, &aop, core->offset,
+			(const ut8*)hex, bufsz);
+		if (ret>0) {
+			str = R_STRBUF_SAFEGET (&aop.esil);
```
Comment (radare) [inline libr/core/cmd_anal.c]: check if str is null

File: libr/core/cmd_anal.c [hunk 9]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+			str2 = calloc(sizeof(char), strlen(str)+1);
+			strcat(str2," ");
```
Comment (radare) [inline libr/core/cmd_anal.c]: missing spaces

File: libr/core/cmd_anal.c [hunk 10]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+
+		if (!r_core_process_input_pade (core, input+pos, &hex, &new_arch, &new_bits)) {
+			// XXX - print help message
+			//return false;
+		}
+
+		if (new_arch == NULL) new_arch = strdup (old_arch);
+		if (new_bits == -1) new_bits = old_bits;
+
+		if (strcmp (new_arch, old_arch) != 0 || new_bits != old_bits){
```
Comment (radare) [inline libr/core/cmd_anal.c]: != 0 is implicit

File: libr/core/cmd_anal.c [hunk 11]
```diff
@@ -2281,6 +2281,48 @@ static void cmd_anal_esil(RCore *core, const char *input) {
...
+		if (input[0])
+			for (pos = 1; pos < R_BIN_SIZEOF_STRINGS && input[pos]; pos++)
+				if (input[pos] == ' ') break;
+
+		if (!r_core_process_input_pade (core, input+pos, &hex, &new_arch, &new_bits)) {
+			// XXX - print help message
+			//return false;
+		}
+
+		if (new_arch == NULL) new_arch = strdup (old_arch);
```
Comment (radare) [inline libr/core/cmd_anal.c]: use ! instead of == NULL

**Comment by radare** [comment]
You can send a PR fixing those syntactic issues, so you will get in touch with the syntax rules we try to follow. Thanks!

**Pull request closed without merging**